irohad/model/converters/impl/json_query_factory.cpp

```diff
@@ -105,136 +115,178 @@ namespace iroha {
-        pb_query.set_creator_account_id(
-            obj_query["creator_account_id"].GetString());
-
-        // set query counter
-        pb_query.set_query_counter(obj_query["query_counter"].GetUint64());
-
-
         auto query_type = obj_query["query_type"].GetString();
-
         auto it = deserializers_.find(query_type);
-        if (it != deserializers_.end() and
-            (this->*it->second)(obj_query, pb_query)) {
-          return pb_query;
-        } else {
-          return nonstd::nullopt;
+        if (it != deserializers_.end()) {
+          auto query = (this->*it->second)(obj_query);
+          if (not query) {
+            return nonstd::nullopt;
+          }
+
+          // Set query signature
+          Signature signature;
+          hexstringToArray(sig["pubkey"].GetString(), signature.pubkey);
+          hexstringToArray(sig["signature"].GetString(), signature.signature);
+          query->signature = signature;
+          return query;
         }
+        log_->error("No query type found");
+        return nonstd::nullopt;
       }
 
-      bool JsonQueryFactory::deserializeGetAccount(
-          rapidjson::GenericValue<rapidjson::UTF8<char>>::Object &obj_query,
-          protocol::Query &pb_query) {
+      std::shared_ptr<iroha::model::Query>
+      JsonQueryFactory::deserializeGetAccount(
+          rapidjson::GenericValue<rapidjson::UTF8<char>>::Object &obj_query) {
         if (not obj_query.HasMember("account_id")) {
-          log_->error("No account_id in json");
-          return false;
+          log_->error("No account id in json");
+          return nullptr;
         }
         if (not obj_query["account_id"].IsString()) {
           log_->error("Type mismatch account_id in json");
-          return false;
+          return nullptr;
         }
-        auto pb_get_account = pb_query.mutable_get_account();
-        pb_get_account->set_account_id(obj_query["account_id"].GetString());
-
-        return true;
+        auto timestamp = obj_query["created_ts"].GetUint64();
+        auto creator = obj_query["creator_account_id"].GetString();
+        auto counter = obj_query["query_counter"].GetUint64();
+        auto account_id = obj_query["account_id"].GetString();
+        return query_generator_.generateGetAccount(timestamp, creator, counter,
+                                                   account_id);
       }
 
-      bool JsonQueryFactory::deserializeGetSignatories(
-          rapidjson::GenericValue<rapidjson::UTF8<char>>::Object &obj_query,
-          protocol::Query &pb_query) {
+      std::shared_ptr<iroha::model::Query>
+      JsonQueryFactory::deserializeGetSignatories(
+          rapidjson::GenericValue<rapidjson::UTF8<char>>::Object &obj_query) {
         if (not obj_query.HasMember("account_id")) {
           log_->error("No account id in json");
-          return false;
+          return nullptr;
         }
         if (not obj_query["account_id"].IsString()) {
```

Comment:
May be better to move those checking in one place? 
Looks for [this](https://gist.github.com/muratovv/15dfe3385696386ec2e2e99050524c59) scheme.